src/main/java/matsyir/pvpperformancetracker/controllers/FightPerformance.java

```diff
@@ -489,4 +503,114 @@ public int compareTo(FightPerformance o)
 		return diff == 0 ? 0 :
 			(int)(diff / Math.abs(diff));
 	}
+
+	/**
+	 * Calculates robe hits for both competitors based on defender gear in each log entry and the configured filter.
+	 */
+	public void calculateRobeHits(PvpPerformanceTrackerConfig.RobeHitFilter filter)
+	{
+		competitorRobeHits = 0;
+		opponentRobeHits = 0;
+		if (filter == null || getAllFightLogEntries() == null || getAllFightLogEntries().isEmpty())
+		{
+			return;
+		}
+
+		for (FightLogEntry entry : getAllFightLogEntries())
+		{
+			// Only consider melee and ranged attacks for robe hits
+			AnimationData ad = entry.getAnimationData();
+			if (ad == null)
+			{
+				continue;
+			}
+			AttackStyle style = ad.attackStyle;
+			if (style == AttackStyle.MAGIC)
+			{
+				continue;
+			}
+
+			if (entry == null)
+			{
```

Comment:
Minor thing, but this entry == null check is pointless the way it's structured currently - if it's null we'd get an error when trying to use the entry above. I'll move it to be the first if in the for loop. I also don't think these entries can ever really be null, but doesn't hurt to have the check.